commons-numbers-complex-arrays/src/test/java/org/apache/commons/numbers/complex/arrays/ComplexListTest.java

```diff
@@ -0,0 +1,220 @@
+
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.set(-2, Complex.ofCartesian(200, 1)));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.set(size, Complex.ofCartesian(200, 1)));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.set(size + 1, Complex.ofCartesian(200, 1)));
+    }
+
+    @Test
+    void testAddIndexOutOfBoundExceptions() {
+        ComplexList list = new ComplexList();
+        int size = 5;
+        IntStream.range(0, size).mapToObj(i -> Complex.ofCartesian(i, -i)).forEach(list::add);
+
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () ->
+            list.add(-1, Complex.ofCartesian(42, 13)));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () ->
+            list.add(size + 1, Complex.ofCartesian(42, 13)));
+    }
+
+    @Test
+    void testRemoveIndexOutOfBoundExceptions() {
+        ComplexList list = new ComplexList();
+        list.add(Complex.ofCartesian(42, 13));
+        list.addAll(list);
+        list.remove(0);
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.remove(1));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.remove(-1));
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {0, 1, 10})
+    void testConstructor(int size) {
+        List<Complex> l1 = new ArrayList<>(size);
+        List<Complex> l2 = new ComplexList(size);
+        Assertions.assertEquals(l1, l2);
+        assertListOperation(l -> l.add(Complex.ofCartesian(10, 20)), l1, l2);
+        assertListOperation(l -> {
+            l.add(1, Complex.ofCartesian(10, 20));
+            return Boolean.TRUE;
+        }, l1, l2);
+        assertListOperation(l -> l.addAll(1, l), l1, l2);
+    }
+
+    @Test
+    void testCapacityExceptions() {
+
+        Assertions.assertThrows(IllegalArgumentException.class, () -> new ComplexList(ComplexList.MAX_CAPACITY + 1));
+
+        // Set-up required sizes
+        ComplexList list = new ComplexList();
+        List<Complex> l = new SizedList(Integer.MAX_VALUE);
+        Assertions.assertThrows(OutOfMemoryError.class, () -> list.addAll(l));
+
+        List<Complex> l2 = new SizedList(ComplexList.MAX_CAPACITY + 1);
+        Assertions.assertThrows(OutOfMemoryError.class, () -> list.addAll(l2));
+    }
+
+    private static <T> void assertListOperation(Function<List<Complex>, T> operation,
+                                                List<Complex> l1, List<Complex> l2) {
+        T t1 = operation.apply(l1);
+        T t2 = operation.apply(l2);
+        Assertions.assertEquals(t1, t2);
+        Assertions.assertEquals(l1, l2);
+    }
+
+    private static <T> void assertListOperation(Function<List<Complex>, T> operation) {
+        assertListOperation(operation, new ArrayList<>(), new ComplexList());
+    }
+
+    private static class SizedList extends ArrayList<Complex> {
```

Comment:
This should be commented so its purpose is clear. It deliberately reports a fixed size (and is therefore a non-functional list) and is used to trigger capacity exceptions when adding a collection to the ComplexList.